jobsdb/jobsdb.go

```diff
@@ -1032,6 +1022,13 @@ func (jd *Handle) readerSetup(ctx context.Context, l lock.LockToken) {
 	// Even if two different services (gateway and processor) perform this operation, there should not be any problem.
 	jd.recoverFromJournal(ReadWrite)
 	jd.assertError(jd.doRefreshDSRangeList(l))
+	jd.assertError(func() error {
+		err := jd.doCleanup(ctx)
```

Comment:
I would prefer this step to be performed in it's own go routine and not block the startup process for long time.  We did see startup delays if there are lot of pending tables.

Reply:
The way it's written now, it needs a dsListLock(the token passed), so running it async would lead to synchronisation issues.
And this is something that doesn't happen often, and stops when it encounters a DS without any old jobs - so it felt safe to run at startup.

Reply:
> And this is something that doesn't happen often, and stops when it encounters a DS without any old jobs
Alright! cool